src/FourBitTwoDisclosureDeviceUnlocker.java

```diff
@@ -55,7 +64,10 @@
      *         device are now identical); false otherwise
      */
     public static boolean unlock(final Device dev) {
-        if (dev != null) {
+        if (dev == null) {
+            return false;
+        }
+        else {
```

Comment:
CheckStyle gave a suggestion to make this less confusing by making the if conditional check for an equality rather than an inequality.